### PR TITLE
Bump react-scripts from 3.4.1 to 4.0.3 + fix associated tech debt

### PR DESCRIPTION
**Pull request by 0xdevalias**
Bumps [react-scripts](https://github.com/facebook/create-react-app/tree/master/packages/react-scripts) from 3.4.1 to 4.0.3.

Partially fixes sparkletown/internal-sparkle-issues#617

Partially fixes sparkletown/internal-sparkle-issues#645

### Release Notes

- https://github.com/facebook/create-react-app/releases/tag/v4.0.3
  - > v4.0.3 is a maintenance release that includes minor bug fixes and dependency updates.
- https://github.com/facebook/create-react-app/releases/tag/v4.0.2
  - > v4.0.2 is a maintenance release that includes minor bug fixes and documentation updates.
- https://github.com/facebook/create-react-app/releases/tag/v4.0.1
  - > v4.0.1 is a maintenance release that includes minor bug fixes and documentation updates.
- https://github.com/facebook/create-react-app/releases/tag/v4.0.0
  - > Create React App 4.0 is a major release with several new features, including support for Fast Refresh!
  - > NOTE: You may need to delete your `node_modules` folder and reinstall your dependencies by running yarn (or npm install) if you encounter errors after upgrading.
  - Highlights
    - Fast Refresh
    - React 17 support
      - New JSX transform
    - TypeScript 4 support
    - ESLint 7
      - New Jest and React Testing Library rules
    - Jest 26
    - PWA/workbox improvements
      - Switch to the Workbox InjectManifest plugin
      - Now its own template so it can be released independently
    - Web Vitals support
- https://github.com/facebook/create-react-app/releases/tag/v3.4.4
  - > v3.4.4 release bumps `resolve-url-loader` to a version for which npm audit does not report a vulnerability. Note that this vulnerability did not affect Create React App
  -  projects, so this change is only necessary to satisfy auditing tools.
- https://github.com/facebook/create-react-app/releases/tag/v3.4.3
  - > v3.4.3 release bumps `terser-webpack-plugin` to a version for which npm audit does not report a vulnerability. Note that this vulnerability did not affect Create React App projects, so this change is only necessary to satisfy auditing tools.
- https://github.com/facebook/create-react-app/releases/tag/v3.4.2
  - > v3.4.2 release bumps `webpack-dev-server` to a version for which npm audit does not report a vulnerability. Note that this vulnerability did not affect Create React App projects, so this change is only necessary to satisfy auditing tools.
  
## Notes

- ~We will probably need to land this `react-scripts` update before we can upgrade `node-sass` in https://github.com/sparkletown/sparkle/pull/1079~
  - ~see https://github.com/sparkletown/sparkle/pull/1079#issuecomment-761886726 / https://github.com/sparkletown/sparkle/issues/850~
  - Already handled: https://github.com/sparkletown/sparkle/pull/1103
- We probably want to disable the new JSX transform for now, as it can have incompatibilities with other tools, and probably warrants it's own standalone PR to enable
  - https://create-react-app.dev/docs/advanced-configuration/
    - `DISABLE_NEW_JSX_TRANSFORM=true`

## TODO

- [x] Rebase onto `staging` once https://github.com/sparkletown/sparkle/pull/1103 has been merged
- [x] Will any of our eslint plugins/etc need to be updated for eslint 7.x?
- [x] Anything else?

### CODE REVIEW

**Comment by 0xdevalias** [comment]
Note: react-scripts is up to `v4.0.3` now, so when we come back back to this, we should make sure to target at least that version (or whatever is then the latest):

- https://github.com/facebook/create-react-app/releases/tag/v4.0.3

**Comment by 0xdevalias** [comment]
```
Failed to compile.

./src/components/molecules/NavBar/NavBar.scss (./node_modules/css-loader/dist/cjs.js??ref--5-oneOf-6-1!./node_modules/postcss-loader/src??postcss!./node_modules/resolve-url-loader??ref--5-oneOf-6-3!./node_modules/sass-loader/dist/cjs.js??ref--5-oneOf-6-4!./src/components/molecules/NavBar/NavBar.scss)
Error: Can't resolve '../../../icons/sparkle-nav-logo.png' in '/Users/devalias/dev/sparkle/sparkle/src/components/molecules/NavBar'
```

- https://github.com/facebook/create-react-app/issues/9870
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-715562244
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-716096895
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-717071085
    - > It looks like this is caused by the css-loader migration to v4
    - > Originally, a root-relative path like /foo/bar.ext would remain untouched, but as per css-loader@4 it becomes processed
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-719666170
    - > If you care to dive into the react-scripts webpack setup, you can pass in the option to disable url routing to css-loader. In node_modules/react-scripts/config/webpack.config.js on line 121 
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-719823931
    - > Looking at css-loader it looks like they're not going to fix it: https://github.com/webpack-contrib/css-loader/issues/1136#issuecomment-664984703
    - > You would have to include /public to the resolve paths. I just moved my assets to /src as I didn't want to eject
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-740754733
    - > Here's a fix using react-app-rewired for css files with absolute links.
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-762766356
    - > As developer of css-loader, I want to say that css-loader supports /images/image.png in url, CRA should add src to resolver.roots (and using /images/image.png will be resolved as /src/images/image.png, it is default behavior for webpack@5)
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-778349197
    - > I moved svg, png, jpg and font files from /public to /src and referenced them in jsx/scss/css files as recommended here:
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-791428022
    - > Just note, the latest version of css-loader supports /* webpackIgnore: true */, i.e. you can write
  - https://github.com/facebook/create-react-app/issues/9870#issuecomment-813003970
    - > Moving assets (at least those read by css-loader) from public to src is not a big deal and is the "recommended way" of Adding Images, Fonts, and Files as stated in the CRA documentation: create-react-app.dev/docs/adding-images-fonts-and-files

**Comment by 0xdevalias** [comment]
```
Starting the development server...

=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <3.10.0

YOUR TYPESCRIPT VERSION: 4.2.4

Please only submit bug reports when using the officially supported version.

=============
```

**Comment by 0xdevalias** [comment]
See diff of changes in https://github.com/sparkletown/sparkle/pull/1102/commits/031b1bdfd1245dea82185b748336878c85c6722c:

```
⇒  npm dedup
removed 76 packages, moved 18 packages and audited 3253 packages in 26.929s
```

This seems to have fixed the issues described in https://github.com/sparkletown/sparkle/pull/1102#issuecomment-828182742 without having to delete/re-create the entire `package-lock.json` file (https://github.com/facebook/create-react-app/issues/9083#issuecomment-828309797):

```
⇒  npx eslint --print-config src/index.tsx
{
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "jest": true,
    "node": true
  },
  "globals": {},
  "parser": "/Users/devalias/dev/sparkle/sparkle/node_modules/@typescript-eslint/parser/dist/index.js",
// ..snip..
```

**Comment by 0xdevalias** [comment]
After https://github.com/sparkletown/sparkle/pull/1102/commits/4c651efd7e91365e60b77fbe5f50639ecf3867e4, the following are now being reported:

```
⇒  npm run eslint:check

> co-reality-map@0.1.0 eslint:check /Users/devalias/dev/sparkle/sparkle
> eslint src scripts functions --ext .js,.jsx,.ts,.tsx


/Users/devalias/dev/sparkle/sparkle/functions/venue.js
   73:18  warning  'checkUserIsOwner' was used before it was defined  @typescript-eslint/no-use-before-define
  714:8   warning  'dataOrUpdateKey' was used before it was defined   @typescript-eslint/no-use-before-define
  715:8   warning  'dataOrUpdateKey' was used before it was defined   @typescript-eslint/no-use-before-define
  722:23  warning  'dataOrUpdateKey' was used before it was defined   @typescript-eslint/no-use-before-define
  727:17  warning  'dataOrUpdateKey' was used before it was defined   @typescript-eslint/no-use-before-define

/Users/devalias/dev/sparkle/sparkle/src/components/organisms/ProfileModal/EditProfileForm/EditProfileForm.tsx
  14:11  error  'EditProfileFormValuesType' is defined but never used  @typescript-eslint/no-unused-vars

/Users/devalias/dev/sparkle/sparkle/src/components/templates/Audience/Audience.tsx
  224:9  warning  The 'partygoersBySeat' array makes the dependencies of useMemo Hook (at line 483) change on every render. To fix this, wrap the initialization of 'partygoersBySeat' in its own useMemo() Hook  react-hooks/exhaustive-deps

/Users/devalias/dev/sparkle/sparkle/src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx
  50:9  warning  The 'venueEvents' logical expression could make the dependencies of useMemo Hook (at line 68) change on every render. Move it inside the useMemo callback. Alternatively, wrap the initialization of 'venueEvents' in its own useMemo() Hook  react-hooks/exhaustive-deps

/Users/devalias/dev/sparkle/sparkle/src/components/templates/Playa/AvatarLayer.tsx
  193:9  warning  The 'menu' object makes the dependencies of useMemo Hook (at line 292) change on every render. To fix this, wrap the initialization of 'menu' in its own useMemo() Hook  react-hooks/exhaustive-deps

/Users/devalias/dev/sparkle/sparkle/src/hooks/useConnectRelatedVenues.ts
  121:9  warning  The 'subvenues' logical expression could make the dependencies of useCallback Hook (at line 157) change on every render. To fix this, wrap the initialization of 'subvenues' in its own useMemo() Hook          react-hooks/exhaustive-deps
  121:9  warning  The 'subvenues' logical expression could make the dependencies of useMemo Hook (at line 266) change on every render. To fix this, wrap the initialization of 'subvenues' in its own useMemo() Hook              react-hooks/exhaustive-deps
  124:9  warning  The 'siblingVenues' logical expression could make the dependencies of useCallback Hook (at line 148) change on every render. To fix this, wrap the initialization of 'siblingVenues' in its own useMemo() Hook  react-hooks/exhaustive-deps
  124:9  warning  The 'siblingVenues' logical expression could make the dependencies of useMemo Hook (at line 266) change on every render. To fix this, wrap the initialization of 'siblingVenues' in its own useMemo() Hook      react-hooks/exhaustive-deps

/Users/devalias/dev/sparkle/sparkle/src/hooks/useFirestoreConnect.ts
  6:3  error  'useFirestoreConnect' import from 'react-redux-firebase' is restricted. Import typed version from hooks/useSparkleFirestoreConnect  no-restricted-imports

/Users/devalias/dev/sparkle/sparkle/src/pages/Account/Venue/VenueMapEdition/CustomDragLayer.tsx
  1:17  error  'CSSProperties' is defined but never used  @typescript-eslint/no-unused-vars

/Users/devalias/dev/sparkle/sparkle/src/pages/Admin/AdminVenuePreview.tsx
  26:14  warning  'AdminVenuePreview' is already defined  @typescript-eslint/no-redeclare

/Users/devalias/dev/sparkle/sparkle/src/pages/Admin/Venue/Rooms/RoomsForm.tsx
  106:7  warning  'RoomInnerForm' is already defined  @typescript-eslint/no-redeclare

/Users/devalias/dev/sparkle/sparkle/src/serviceWorker.js
  39:9  warning  'checkValidServiceWorker' was used before it was defined  @typescript-eslint/no-use-before-define
  51:9  warning  'registerValidSW' was used before it was defined          @typescript-eslint/no-use-before-define

/Users/devalias/dev/sparkle/sparkle/src/types/reactions.ts
  16:13  warning  'TextReactionType' is already defined  @typescript-eslint/no-redeclare

/Users/devalias/dev/sparkle/sparkle/src/utils/selectors.ts
  226:31  error  'T' is defined but never used  @typescript-eslint/no-unused-vars
  231:36  error  'T' is defined but never used  @typescript-eslint/no-unused-vars
  250:36  error  'T' is defined but never used  @typescript-eslint/no-unused-vars

✖ 23 problems (6 errors, 17 warnings)
```

Need to check through these, ensure they aren't false positives, and then fix up whats left.

**Comment by 0xdevalias** [comment]
Following on from https://github.com/sparkletown/sparkle/pull/1102#issuecomment-828327448, we now have the following issues left to resolve:

```
⇒  npm run eslint:check

> co-reality-map@0.1.0 eslint:check /Users/devalias/dev/sparkle/sparkle
> eslint src scripts functions --ext .js,.jsx,.ts,.tsx


/Users/devalias/dev/sparkle/sparkle/src/components/templates/Audience/Audience.tsx
  224:9  warning  The 'partygoersBySeat' array makes the dependencies of useMemo Hook (at line 483) change on every render. To fix this, wrap the initialization of 'partygoersBySeat' in its own useMemo() Hook  react-hooks/exhaustive-deps

/Users/devalias/dev/sparkle/sparkle/src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx
  50:9  warning  The 'venueEvents' logical expression could make the dependencies of useMemo Hook (at line 68) change on every render. Move it inside the useMemo callback. Alternatively, wrap the initialization of 'venueEvents' in its own useMemo() Hook  react-hooks/exhaustive-deps

/Users/devalias/dev/sparkle/sparkle/src/hooks/useConnectRelatedVenues.ts
  121:9  warning  The 'subvenues' logical expression could make the dependencies of useCallback Hook (at line 157) change on every render. To fix this, wrap the initialization of 'subvenues' in its own useMemo() Hook          react-hooks/exhaustive-deps
  121:9  warning  The 'subvenues' logical expression could make the dependencies of useMemo Hook (at line 266) change on every render. To fix this, wrap the initialization of 'subvenues' in its own useMemo() Hook              react-hooks/exhaustive-deps
  124:9  warning  The 'siblingVenues' logical expression could make the dependencies of useCallback Hook (at line 148) change on every render. To fix this, wrap the initialization of 'siblingVenues' in its own useMemo() Hook  react-hooks/exhaustive-deps
  124:9  warning  The 'siblingVenues' logical expression could make the dependencies of useMemo Hook (at line 266) change on every render. To fix this, wrap the initialization of 'siblingVenues' in its own useMemo() Hook      react-hooks/exhaustive-deps

✖ 6 problems (0 errors, 6 warnings)
```

**Comment by 0xdevalias** [comment]
Following on from https://github.com/sparkletown/sparkle/pull/1102#issuecomment-829897794, it seems as though we've fixed all of the outstanding eslint issues now!

**Comment by 0xdevalias** [comment]
Following on from https://github.com/sparkletown/sparkle/pull/1102#issuecomment-829897794, it seems as though we've fixed all of the outstanding eslint issues now!

```
⇒  npm run eslint:check

> co-reality-map@0.1.0 eslint:check /Users/devalias/dev/sparkle/sparkle
> eslint src scripts functions --ext .js,.jsx,.ts,.tsx
```

File: functions/venue.js
Comment (0xdevalias) [inline functions/venue.js:68]: [note] The changes in this file _should_ just be moving the position of the functions around, there shouldn't be any changes to the code within them.

File: package.json
Comment (0xdevalias) [inline package.json:151]: [note] These are all provided by `react-scripts`, so there is no need for us to specify them directly here like this (and doing so can lead to weird errors/version mismatches/etc)

File: src/components/organisms/ProfileModal/EditProfileForm/EditProfileForm.tsx
Comment (0xdevalias) [inline src/components/organisms/ProfileModal/EditProfileForm/EditProfileForm.tsx:18]: [note] This was unused.

File: src/hooks/useFirestoreConnect.ts
Comment (0xdevalias) [inline src/hooks/useFirestoreConnect.ts:8]: [note] This is the one place where we actually want this direct import allowed, thus disabling the lint warning.

File: src/serviceWorker.js
Comment (0xdevalias) [inline src/serviceWorker.js:20]: [note] The changes in this file _should_ just be moving the position of the functions around, there shouldn't be any changes to the code within them.

File: tsconfig.json
Comment (0xdevalias) [inline tsconfig.json:22]: [note] This was added automatically after we upgraded `react-scripts` the first time we `npm start`'d

File: package.json
Comment (0xdevalias) [inline package.json:62]: [note] This was the main intended change in this PR, the rest is related to changed dependencies, lint rules, etc because of this.

**Review by 0xdevalias — COMMENTED** [review]
[note] Some notes to explain various aspects of this PR to make reviewing easier.

Note also that the mass of changes to `package-lock.diff` are due to `react-scripts` (and all of the dependencies it wraps up) changing version (as we intended), as can be seen in https://github.com/sparkletown/sparkle/pull/1102/commits/dbeb3944cc342f3cf5d33a1c7f16315bcc87c1cd

**Comment by 0xdevalias** [comment]
## Before

```
⇒  npm i
npm WARN eslint-config-react-app@5.2.1 requires a peer of @typescript-eslint/eslint-plugin@2.x but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-react-app@5.2.1 requires a peer of @typescript-eslint/parser@2.x but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-react-app@5.2.1 requires a peer of eslint-plugin-react-hooks@1.x || 2.x but none is installed. You must install peer dependencies yourself.
npm WARN The package @types/styled-components is included as both a dev and production dependency.

audited 2827 packages in 9.807s

91 packages are looking for funding
  run `npm fund` for details

found 5137 vulnerabilities (5097 low, 5 moderate, 35 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

```
⇒  npm run build

> co-reality-map@0.1.0 build /Users/devalias/dev/sparkle/sparkle
> react-scripts build

Creating an optimized production build...
Compiled successfully.

File sizes after gzip:

  628.17 KB  build/static/js/2.2a99c644.chunk.js
  107.98 KB  build/static/js/main.2df18998.chunk.js
  97.27 KB   build/static/css/main.aade0aa4.chunk.css
  9.45 KB    build/static/js/3.c0da807d.chunk.js
  1.17 KB    build/static/js/runtime-main.d2e9ab6b.js
```

image

## After

```
⇒  npm i
npm WARN @babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining@7.13.12 requires a peer of @babel/core@^7.13.0 but none is installed. You must install peer dependencies yourself.
npm WARN The package @types/styled-components is included as both a dev and production dependency.

audited 3319 packages in 11.484s

166 packages are looking for funding
  run `npm fund` for details

found 471 vulnerabilities (434 low, 1 moderate, 36 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

```
⇒  npm run build

> co-reality-map@0.1.0 build /Users/devalias/dev/sparkle/sparkle
> react-scripts build

Creating an optimized production build...
Compiled successfully.

File sizes after gzip:

  630.06 KB  build/static/js/2.d00b77b8.chunk.js
  130.2 KB   build/static/css/main.9b7a5c6f.chunk.css
  117.31 KB  build/static/js/main.44653ba0.chunk.js
  9.49 KB    build/static/js/3.1e276629.chunk.js
  1.17 KB    build/static/js/runtime-main.d0e24706.js
```

**Note:** I believe the slight increase in size shown in this bundle is due to moving some of the image assets from `public/` (where they won't have been reported in this bundle size) to being properly included as part of the bundle (using the recommended/best practice method of 'importing' the image files in the code, rather than referencing them with 'magic strings')

image

**Comment by 0xdevalias** [comment]
https://create-react-app.dev/docs/supported-browsers-features/

> Note that this project includes no polyfills by default.
> 
> If you use any other ES6+ features that need runtime support (such as Array.from() or Symbol), make sure you are including the appropriate polyfills manually, or that the browsers you are targeting already support them.

---

https://create-react-app.dev/docs/supported-browsers-features/#configuring-supported-browsers

> The browserslist configuration controls the outputted JavaScript so that the emitted code will be compatible with the browsers specified. The production list will be used when creating a production build by running the build script, and the development list will be used when running the start script. You can use https://browserl.ist to see the browsers supported by your configured browserslist.
>
> > Note that this does not include polyfills automatically for you. You will still need to polyfill language features (see above) as needed based on the browsers you are supporting.

---

https://github.com/facebook/create-react-app/blob/master/packages/react-app-polyfill/README.md

This package includes polyfills for various browsers. It includes minimum requirements and commonly used language features used by Create React App projects.

https://github.com/facebook/create-react-app/blob/master/packages/react-app-polyfill/README.md#polyfilling-other-language-features

> You can also polyfill stable language features not available in your target browsers. If you're using this in Create React App, it will automatically use the browserslist you've defined to only include polyfills needed by your target browsers when importing the stable polyfill. Make sure to follow the Internet Explorer steps above if you need to support Internet Explorer in your application.
>
> ```typescript
> // This must be the first line in src/index.js
> import 'react-app-polyfill/stable';
> ```
>
> If you are supporting Internet Explorer 9 or Internet Explorer 11 you should include both the ie9 or ie11 and stable modules:
>
> ```
> // These must be the first lines in src/index.js
> import 'react-app-polyfill/ie11';
> import 'react-app-polyfill/stable';
> ```

We seem to already have this via `react-scripts`:

```
⇒  npm ls react-app-polyfill
co-reality-map@0.1.0 /Users/devalias/dev/sparkle/sparkle
└─┬ react-scripts@4.0.3
  └── react-app-polyfill@2.0.0
```

---

- https://github.com/sparkletown/sparkle/blob/staging/src/polyfills.ts

---

https://babeljs.io/docs/en/babel-preset-env.html

> `@babel/preset-env` is a smart preset that allows you to use the latest JavaScript without needing to micromanage which syntax transforms (and optionally, browser polyfills) are needed by your target environment(s). This both makes your life easier and JavaScript bundles smaller!

This seems to already be included via `react-scripts -> babel-preset-react-app`:

```
⇒  npm ls @babel/preset-env

co-reality-map@0.1.0 /Users/devalias/dev/sparkle/sparkle
└─┬ react-scripts@4.0.3
  ├─┬ @svgr/webpack@5.5.0
  │ └── @babel/preset-env@7.13.15
  ├─┬ babel-preset-react-app@10.0.0
  │ └── @babel/preset-env@7.12.1
  └─┬ workbox-webpack-plugin@5.1.4
    └─┬ workbox-build@5.1.4
      └── @babel/preset-env@7.13.15  deduped
```

https://babeljs.io/docs/en/babel-preset-env.html#bugfixes

> Note: These optimizations will be enabled by default in Babel 8
> 
> By default, @babel/preset-env (and Babel plugins in general) grouped ECMAScript syntax features into collections of closely related smaller features. These groups can be large and include a lot of edge cases, for example "function arguments" includes destructured, default and rest parameters. From this grouping information, Babel enables or disables each group based on the browser support target you specify to @babel/preset-env’s targets option.
> 
> When this option is enabled, @babel/preset-env tries to compile the broken syntax to the closest non-broken modern syntax supported by your target browsers. Depending on your targets and on how many modern syntax you are using, this can lead to a significant size reduction in the compiled app. This option merges the features of @babel/preset-modules without having to use another preset.

https://babeljs.io/docs/en/babel-preset-env.html#usebuiltins

> This option configures how @babel/preset-env handles polyfills.
> 
> When either the usage or entry options are used, @babel/preset-env will add direct references to core-js modules as bare imports (or requires). This means core-js will be resolved relative to the file itself and needs to be accessible.
> 
> Since @babel/polyfill was deprecated in 7.4.0, we recommend directly adding core-js and setting the version via the corejs option.

https://babeljs.io/docs/en/babel-preset-env.html#corejs

> This option only has an effect when used alongside useBuiltIns: usage or useBuiltIns: entry, and ensures @babel/preset-env injects the polyfills supported by your core-js version. It is recommended to specify the minor version otherwise "3" will be interpreted as "3.0" which may not include polyfills for the latest features.
> 
> By default, only polyfills for stable ECMAScript features are injected: if you want to polyfill proposals, you have three different options:

File: src/components/molecules/NavSearchBar/NavSearchBar.tsx
Comment (mike-lvov) [inline src/components/molecules/NavSearchBar/NavSearchBar.tsx:153]: Wouldn't it be nicer to remove it & use font awesome instead?

File: src/components/templates/Audience/Audience.tsx
Comment (mike-lvov) [inline src/components/templates/Audience/Audience.tsx:308]: I guess we use null here because the implementation with undefined is not merged?

**Review by mike-lvov — APPROVED** [review]
A few notes/questions

File: src/components/molecules/NavSearchBar/NavSearchBar.tsx
Comment (0xdevalias) [reply src/components/molecules/NavSearchBar/NavSearchBar.tsx:153]: Potentially, though @mcflyok tends not to like the font awesome iconography in general I believe.

This was just renaming the variable name of existing code, not intending to change logic in this PR.

File: src/components/templates/Audience/Audience.tsx
Comment (0xdevalias) [reply src/components/templates/Audience/Audience.tsx:308]: Where is the implementation with `undefined`?

I didn't seek to change any of the existing functionality in this PR, so I left it as what it was.

**Review by 0xdevalias — COMMENTED** [review]
Looks good to me (let's see if this satisfies the CODEOWNERS rule for @sparkletown/engineering-leads)

---

**Edit:** Seemingly it doesn't, and is a long outstanding issue: https://github.community/t/do-not-require-owner-approval-if-the-pull-request-is-from-an-owner/369/64

Temporarily disabled the 'require CODEOWNERS' branch protection to get around this.

**Pull request merged**
